core/src/main/scala/tofu/syntax/bracket.scala

```diff
@@ -36,8 +36,31 @@ object bracket {
   }
 
   implicit final class TofuBracketMVarOps[F[_], A](private val mvar: MVar[F, A]) extends AnyVal {
-    def bracketUpdate[B, E](use: A => F[B])(implicit F: Applicative[F], FG: Guarantee[F]): F[B] =
-      mvar.take.bracketIncomplete(use)(mvar.put)
+
+    /**
+      * Update value with effectful transformation. In case of error value remains unchanged
+      * @param use function to atomically modify value contained in `MVar`
+      * @return `F[A]` modified value contained in `MVar`
+      */
+    def bracketUpdate(use: A => F[A])(implicit FG: Guarantee[F], M: Monad[F]): F[A] =
+      M.flatMap(mvar.take)(oldA =>
+        M.pure(oldA).bracketOpt(use) { (newA, success) => mvar.put(if (success) newA else oldA) }
+      )
+
+    /**
+      * Modify value with effectful transformation, calculating result. In case of error value remains unchanged
+      * @param use function to atomically modify value contained in `MVar` and produce result
+      * @return `F[B]`
+      */
+    // NOTE: maybe return F[(A, B)] to inspect updated mvar?
```

Comment:
notice the note :)

Reply:
If user wants some information from `A` they obviously may use appropriate `B`.